### DISCUSSION TITLE
artifactUrl having some extra characters in OCI environment

### DISCUSSION THREAD

**Discussion by pallakartheekreddy** [Sunbird-Knowlg/Community#179] [Bugs]
@ddevadat
 
We observed In OCI DEV, we noticed that after uploading a PDF, there were additional keys such as `?16\` or `?32\` in the URL during playback. We encountered a similar issue in the production environment. Do you happen to recall the resolution we implemented for this at that time? If so, could you share the details or any insights that might help us address the current occurrence of the same issue?
 
 ```
 https://ax2cel5zyviy.compat.objectstorage.ap-hyderabad-1.oraclecloud.com/sunbird-content-dev/content/assets/do_11393089031132774419/test.pdf?32\
 ```
 
 @gandham-santhosh @mohitga @santhosh-tg @ashokreddy1208

**Comment by damithkothalawala** [comment]
Hi @pallakartheekreddy 

Bro, this is happening due to viewer's js trying to fetch latest file out, thinking backend as CDN. 

Currently oci compact object storage api do not support to handle and ignore invalid url query strings.

On production, we use a CDN to front compact object storage. And we have added a rule at CDN to drop any query strings to avoid that.

For dev, What I have done is that using my OCI free tier instance (0 cost) as a Nginx instance to drop unwanted query string from the url. So that oci compact store api doesn't throw errors.

Here is my sample http config (just run certbot to generate new ssl and auto set this to https)

Hope this helps :-) 

server {
    listen 80;
    server_name http://example.com;

    location / {
        # remove any query string from the request
        set $request_uri_without_query $request_uri;
        if ($request_uri ~ "^(.+)\?(.*)$") {
            set $request_uri_without_query $1;
        }

        # proxy pass to the origin without the query string
        proxy_pass https://ax2cel5zyviy.compat.objectstorage.ap-hyderabad-1.oraclecloud.com$request_uri_without_query;

        # preserve the original domain name
        proxy_set_header Host $host;

        # add some common headers for proxying
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}

**Reply by gandham-santhosh** [reply]
@damithkothalawala thanks for sharing the details. How does this NGINX end point is called from the SunbirdEd dev instance? Any additional configurations need to be done apart from NGINX setup?

**Reply by damithkothalawala** [reply]
Hi Bro @gandham-santhosh  , Just replacing oracle compact object store url with nginx url should work. And can control all CORS conditions also via nginx if required.

**Reply by gandham-santhosh** [reply]
Thanks @damithkothalawala . Is this configuration change done only for content service? or is it done for any  other service as well?

**Reply by damithkothalawala** [reply]
Dear @gandham-santhosh, so far only this change only required for files loaded at viewer side. Better if dev team can add a feature to disable this random query string addition to end of content url.

**Reply by gandham-santhosh** [reply]
@pallakartheekreddy  can this be done?
